toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -27,8 +27,14 @@ class GoogleBigQueryDataSource(ToucanDataSource):
     )
 
 
-# TODO - check bad return type
-def _define_type(value) -> str:
+BigQueryParam = Union[bigquery.ScalarQueryParameter, bigquery.ArrayQueryParameter]
+
+
+# NOTE: This does not play nicely with dates. They're a bit tricky
+# though, as we'd have to try and parse dates from strings to
+# determine if something is a date or not. Until then, we can just
+# use a cast. eg: SELECT * FROM table WHERE STRING(date_col) IN UNNEST({{my_dates}})
```

Comment:
Perhaps, this is a larger topic about how to handle dates in data sources parameters. 
Your solution seems fine for now.